src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -1989,8 +2003,13 @@ void
 JvmtiHandshake::execute(JvmtiUnitedHandshakeClosure* hs_cl, jthread target) {
   JavaThread* current = JavaThread::current();
   HandleMark hm(current);
+  oop thread_oop = JNIHandles::resolve_external_guard(target);
+  bool is_virtual = java_lang_VirtualThread::is_instance(thread_oop);
 
-  JvmtiVTMSTransitionDisabler disabler(target);
+  // Target can be virtual or platform thread.
```

Comment:
Thank you for suggestion. Explained comment.
Now, it looks as below:
```
  // Target can be virtual or platform thread.
  // Disable VTMS transition for one thread if it is virtual.
  // Otherwise, disable VTMS transitions for all threads
  // to protect against VTMS transitions in carrier threads.
  // For the latter VTMS transitions are disabled for all threads by several reasons:
  // - carrier threads can mount virtual threads which may cause incorrect behavior
  // - it is good invariant if we can ensure no VTMS transition might happen at some points 
  // - there is no mechanism to disable transitions for a specific carrier threads yet
```